src/Mallos.Searchable.CodeAnalyzer/SearchableSyntaxReceiver.cs

```diff
@@ -0,0 +1,149 @@
+namespace Mallos.Searchable.CodeAnalyzer
+{
+    using Mallos.Searchable.Attributes;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    class GeneratorNodeAttribute
+    {
+        public AttributeSyntax Key { get; }
+
+        public AttributeSyntax Rule { get; }
+
+        public Type RuleType { get; }
+
+        public GeneratorNodeAttribute(AttributeSyntax key, AttributeSyntax rule, Type ruleType)
+        {
+            this.Key = key;
+            this.Rule = rule;
+            this.RuleType = ruleType;
+        }
+
+        public string GetKey()
+        {
+            if (this.Rule == null)
+                return null;
+
+            return this.Key.ArgumentList.Arguments[0].ToString().Replace("\"", "");
+        }
+    }
+
+    class GeneratorNodeAttributeGroup
+    {
+        public MemberDeclarationSyntax Member { get; }
+
+        public List<GeneratorNodeAttribute> Attributes { get; }
+
+        public GeneratorNodeAttributeGroup(MemberDeclarationSyntax member, List<GeneratorNodeAttribute> attributes)
+        {
+            this.Member = member;
+            this.Attributes = attributes;
+        }
+    }
+
+    class GeneratorNode
+    {
+        public AttributeSyntax FilterGenerator { get; }
+
+        public TypeDeclarationSyntax ObjectSyntax { get; }
+
+        public GeneratorNodeAttributeGroup[] Members { get; }
+
+        public string ClassNamespace => "Mallos.Searchable.Test";
+
+        public string ClassName => this.ObjectSyntax.Identifier.ValueText;
+
+        public string TargetClassName { get; }
+
+        public GeneratorNode(
+            AttributeSyntax filterGenerator,
+            TypeDeclarationSyntax objectSyntax,
+            GeneratorNodeAttributeGroup[] members)
+        {
+            this.FilterGenerator = filterGenerator;
+            this.ObjectSyntax = objectSyntax;
+            this.Members = members;
+
+            // Process Attribute
+            var filterGeneratorAttribute = this.FilterGenerator.ArgumentList.Arguments[0];
+            this.TargetClassName = filterGeneratorAttribute.ToString().Replace("\"", "");
+        }
+    }
+
+    class SearchableSyntaxReceiver : ISyntaxReceiver
+    {
+        public List<GeneratorNode> Generators { get; private set; } = new List<GeneratorNode>();
+
+        private readonly List<Type> filterAttributes;
+
+        public SearchableSyntaxReceiver()
+        {
+            this.filterAttributes = ReflectionHelper.AllOfType<FilterBaseAttribute>().ToList();
```

Comment:
It would be nice to be able to remove this, then we don't have to depend on `Mallos.Searchable` project.

But how can it work without using Reflection if we don't know more about the syntax?